Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -681,29 +681,34 @@ public class PBXProjGenerator {
         return "\(carthagePath)/\(platformName)"
     }
 
-    func getAllCarthageDependencies(target: Target, visitedTargets: [String: Bool] = [:]) -> [Dependency] {
-
+    func getAllCarthageDependencies(target: Target) -> [Dependency] {
         // this is used to resolve cyclical target dependencies
-        var visitedTargets = visitedTargets
-        visitedTargets[target.name] = true
-
+        var visitedTargets: Set<String> = []
         var frameworks: [Dependency] = []
 
-        for dependency in target.dependencies {
-            switch dependency.type {
-            case .carthage:
-                frameworks.append(dependency)
-            case .target:
-                let targetName = dependency.reference
-                if visitedTargets[targetName] == true {
-                    return []
-                }
-                if let target = project.getTarget(targetName) {
-                    frameworks += getAllCarthageDependencies(target: target, visitedTargets: visitedTargets)
+        var queue: [Target] = [target]
+        while !queue.isEmpty {
```

Comment:
Is this traversing the same graph, but just doing it in one function instead of recursively?

I think to get this really performant the carthage dependencies for a target could be cached by just running this once for all targets at the beginning, instead of each target going through the whole graph again

Reply:
It is traversing the same dependencies yes. It should have the exact same result.

Reply:
We could do that, I figured that would be a larger enough change that if we could do this fast enough it's not really worth it.

Reply:
@yonaskolb The recursive version is passing `visitedTargets` by value, which I suspect is why the the iterative version is much faster.